kafka-connector/README.md

```diff
@@ -71,28 +61,72 @@ The resulting jar is at target/cps-kafka-connector.jar.
-| cps.subscription | String | REQUIRED (No default) | The name of the subscription to Cloud Pub/Sub. |
-| cps.project | String | REQUIRED (No default) | The project containing the topic from which to pull messages. |
+| cps.subscription | String | REQUIRED (No default) | The name of the subscription to Cloud Pub/Sub, e.g. "sub" for topic "/projects/bar/subscriptions/sub". |
+| cps.project | String | REQUIRED (No default) | The project containing the topic from which to pull messages, e.g. "bar" from above. |
 | kafka.topic | String | REQUIRED (No default) | The topic in Kafka which will receive messages that were pulled from Cloud Pub/Sub. |
 | cps.maxBatchSize | Integer | 100 | The minimum number of messages to batch per pull request to Cloud Pub/Sub. |
 | kafka.key.attribute | String | null | The Cloud Pub/Sub message attribute to use as a key for messages published to Kafka. |
 | kafka.partition.count | Integer | 1 | The number of Kafka partitions for the Kafka topic in which messages will be published to. |
 | kafka.partition.scheme | round_robin, hash_key, hash_value | round_robin | The scheme for assigning a message to a partition in Kafka. The scheme "round_robin" assigns partitions in a round robin fashion, while the schemes "hash_key" and "hash_value" find the partition by hashing the message key and message value respectively. |
 
-##### Sink Connector
+#### Sink Connector
 
 | Config | Value Range | Default | Description |
 |---------------|-------------|-----------------------|------------------------------------------------------------------------------------------------------------------------------------|
-| cps.topic | String | REQUIRED (No default) | The topic in Cloud Pub/Sub to publish to. |
-| cps.project | String | REQUIRED (No default) | The project in Cloud Pub/Sub containing the topic to publish to. |
+| cps.topic | String | REQUIRED (No default) | The topic in Cloud Pub/Sub to publish to, e.g. "foo" for topic "/projects/bar/topics/foo". |
+| cps.project | String | REQUIRED (No default) | The project in Cloud Pub/Sub containing the topic, e.g. "bar" from above. |
 | maxBufferSize | Integer | 100 | The maximum number of messages that can be received for the messages on a topic partition before publishing them to Cloud Pub/Sub. |
+
+#### Schema Support and Data Model
+
+A pubsub message has two main parts: the message body and attributes. The
+message body is a ByteString object that translates well to and from the byte[]
+bodies of Kafka messages. For this reason, we recommend using primitive data
+types where possible to prevent deserializing and reserializing the same message
+body.
+
+Additionally, a Pubsub message size cannot exceed 10MB, so please check
+your broker's message.max.bytes configuration to prevent possible errors.
+
+The sink connector handles the conversion in the following way:
+
+*   For all primitive data types (integers, floats, bytes, and strings), the
```

Comment:
I'm not sure these details necessarily make sense in the user guide. The concept of "primitive data types" applies only to the Connector infrastructure, not to Kafka as a whole. If someone were using the sink connector to get messages from Kafka into Cloud Pub/Sub, they are going to know about the type of the body of the message, e.g., they have published a JSON or Avro message. In the case that they have published a JSON message, then they can choose to use a StringConverter or a JsonConverter for the CPS sink connector. The former would pass the message in as a string, which would then become the body of the CPS message. The latter would semantically deconstruct the JSON message and store it via message attributes. I would use that as the distinction rather than talking about these primitive data types.